src/choices.c

```diff
@@ -83,7 +83,9 @@ merge(struct choice *front, struct choice *back)
 	c = &head;
 
 	while (front != NULL && back != NULL) {
-		if (front->score > back->score) {
+		if (front->score > back->score ||
+                (front->score == back->score &&
+                 strcmp(front->str, back->str) < 0)) {
```

Comment:
I don't know. What would `n` be? The length of the shortest string? Is that what `strcmp` does?